src/System.Private.ServiceModel/tests/Scenarios/SelfHostWcfService/TestResources/EndpointResource.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using WcfTestBridgeCommon;
+
+namespace WcfService.TestResources
+{
+    internal abstract class EndpointResource<ServiceType, ContractType> : IResource where ServiceType : class, ContractType 
+    {
+        private static Dictionary<string, ServiceHost> currentHosts = new Dictionary<string, ServiceHost>();
+        private static object currentHostLock = new object();
+
+        protected abstract string Protocol { get; }
+
+        protected abstract string Address { get; }
+
+        protected abstract string Port { get; }
+
+        public object Put()
+        {
+            ServiceHost host;
+            if (!currentHosts.TryGetValue(Address, out host))
+            {
+                lock (currentHostLock)
+                {
+                    if (!currentHosts.TryGetValue(Address, out host))
+                    {
+                        host = new ServiceHost(typeof(ServiceType));
+                        // URI assumes localhost but that instead should be passed in
+                        host.AddServiceEndpoint(
```

Comment:
Agree should not be localhost.  Can this be the machine name where the resource itself is running?  That would allow the x-plat scenario where Bridge and service are running on a different machine.  Don't block this PR on this issue, we can come back and fix this later.